tests/e2e/conftest.py

```diff
@@ -25,6 +31,163 @@
+            kind="deployment", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        nb_operator_dc = Deployment(
+            **ocp_deployment_obj.get(resource_name="noobaa-operator")
+        )
+        nb_operator_dc.scale(replicas=0)
+        nb_endpoint_dc = Deployment(
+            **ocp_deployment_obj.get(resource_name="noobaa-endpoint")
+        )
+        nb_endpoint_dc.scale(replicas=0)
+        modify_statefulset_replica_count(
+            statefulset_name=constants.NOOBAA_CORE_STATEFULSET, replica_count=0
+        )
+        logger.info("Stopped the noobaa service!")
+
+        # Login to the NooBaa DB pod and cleanup potential database clients to nbcore
+        query = "SELECT pg_terminate_backend (pid) FROM pg_stat_activity WHERE datname = 'nbcore';"
+        noobaa_db_pod.exec_cmd_on_pod(
+            command=f'psql -h 127.0.0.1 -p 5432 -U postgres -c "{query}"'
+        )
+        logger.info("Cleaned up potential database clients to nbcore!")
+
+        # Restore DB from a local folder
+        OCP(namespace=constants.OPENSHIFT_STORAGE_NAMESPACE).exec_oc_cmd(
+            command=f"cp ./mcg.bck {noobaa_db_pod.name}:/tmp/test.db"
+        )
+        noobaa_db_pod.exec_cmd_on_pod(command="pg_restore -d nbcore /tmp/test.db -c")
+        logger.info("Restored DB from the local folder!")
+
+        # Delete secrets and restore them from a local folder.
+        # Please note that verify that there are no errors before you proceed to the next steps.
+        for secret in secrets_obj:
+            secret.delete()
+        logger.info("Deleted current secrets!")
+        for secret in secrets_obj:
+            secret.create()
+        logger.info("Restored old secrets!")
+
+        # Restore MCG reconciliation
+        params = '{"spec": {"multiCloudGateway": {"reconcileStrategy": "manage"}}}'
+        ocp_storagecluster_obj.patch(
+            resource_name="ocs-storagecluster", params=params, format_type="merge"
+        )
+        logger.info("Restored MCG reconcilation!")
+
+        # Start the NooBaa service
+        nb_operator_dc.scale(replicas=1)
+        nb_endpoint_dc.scale(replicas=1)
+        modify_statefulset_replica_count(
+            statefulset_name=constants.NOOBAA_CORE_STATEFULSET, replica_count=1
+        )
+        logger.info("Started noobaa services!")
+
+        # Restart the NooBaa DB pod
+        noobaa_db_pod.delete()
+        logger.info("Restarted noobaa-db pod!")
+
+        # Make sure the testloss bucket doesn't exists and test bucket consists all the data
+        time.sleep(10)
+        try:
+            s3_list_objects_v1(s3_obj=mcg_obj_session, bucketname=testloss_bucket.name)
+        except Exception as e:
+            logger.info(e)
+
+        logger.info(
+            s3_list_objects_v1(s3_obj=mcg_obj_session, bucketname=test_bucket.name)
+        )
+
+    def finalizer():
+        # remove the local copy of ./mcg.bck
+        os.remove("mcg.bck")
```

Comment:
I feel you need to improve the finalizer part, what if the test fails just after step4? noobaa pods will remain in down state. Please handle such things in the finalizer